sdk/typescript/src/keypairs/passkey/keypair.ts

```diff
@@ -0,0 +1,207 @@
+}
+
+export interface PasskeySignOptions {
+	timeout: number; // The timeout for the passkey authentication dialog.
+}
+
+function validateCreateOptions(options: PasskeyCreateOptions) {
+	if (options.timeout <= 0) throw new Error('Timeout must be positive');
+	if (!options.displayName?.trim()) throw new Error('Display name cannot be empty');
+	if (!options.rpName?.trim()) throw new Error('RP name cannot be empty');
+	if (options.timeout > 300000) throw new Error('Timeout cannot exceed 5 minutes');
+}
+
+export interface PasskeyProvider {
+	create(options: PasskeyCreateOptions): Promise<RegistrationCredential>;
+	get(challenge: Uint8Array, timeout: number): Promise<AuthenticationCredential>;
+}
+
+// Default browser implementation
+export class BrowserPasskeyProvider implements PasskeyProvider {
+	async create(options: PasskeyCreateOptions): Promise<RegistrationCredential> {
+		return (await navigator.credentials.create({
+			publicKey: {
+				challenge: new TextEncoder().encode('Create passkey wallet on Sui'),
+				rp: {
+					name: options.rpName,
+				},
+				user: {
+					id: randomBytes(10),
+					name: 'sui-wallet-user',
+					displayName: options.displayName,
+				},
+				pubKeyCredParams: [{ alg: -7, type: 'public-key' }],
+				authenticatorSelection: {
+					authenticatorAttachment: 'cross-platform',
+					residentKey: 'required',
+					requireResidentKey: true,
+					userVerification: 'required',
+				},
+				timeout: options.timeout ?? 60000,
+			},
+		})) as RegistrationCredential;
+	}
+
+	async get(challenge: Uint8Array, timeout: number): Promise<AuthenticationCredential> {
+		return (await navigator.credentials.get({
+			publicKey: {
+				challenge,
+				userVerification: 'required',
+				timeout,
+			},
+		})) as AuthenticationCredential;
+	}
+}
+
+/**
+ * A passkey signer used for signing transactions. This is a client side implementation for [SIP-9](https://github.com/sui-foundation/sips/blob/main/sips/sip-9.md).
+ */
+export class PasskeyKeypair extends Signer {
+	private publicKey: Uint8Array;
+	private provider: PasskeyProvider;
+
+	/**
+	 * Get the key scheme of passkey,
+	 */
+	getKeyScheme(): SignatureScheme {
+		return 'Passkey';
+	}
+
+	/**
+	 * Creates an instance of Passkey signer. It's expected to call the static `getPasskeyInstance` method to create an instance.
+	 * For example:
+	 * ```
+	 * const signer = await PasskeyKeypair.getPasskeyInstance();
+	 * ```
+	 */
+	constructor(publicKey: Uint8Array, provider: PasskeyProvider) {
+		super();
+		if (publicKey.length !== PASSKEY_PUBLIC_KEY_SIZE) {
+			throw new Error(`Invalid public key size: expected ${PASSKEY_PUBLIC_KEY_SIZE} bytes`);
+		}
+		this.publicKey = publicKey;
+		this.provider = provider;
+	}
+
+	/**
+	 * Creates an instance of Passkey signer invoking the passkey from navigator.
+	 */
+	static async getPasskeyInstance(
+		options: PasskeyCreateOptions = {
+			timeout: 60000,
```

Comment:
I am wondering if this options might make more sense as options for the provider.  I am not entirely sure which is better, but I would be curious if options like timeout are not consistent across passkey apis.  Having the provider be the configurable piece, and keeping the generic passkey signer api as simple as possible would make it easier to avoid having to hack around implementation differences between different apis.

Reply:
basically, this method would only take the provider, and all configuration would go through the provider rather than through the signer instance

Reply:
there are configuration that i find its better to be opinionated about such as must use secp256r1 (if other options are given the signature will just be invalid) and required user verification etc.

Reply:
not saying we need to add any additional configuration, just that the thing that takes options should probably be the provider rather than the signer directly.  cc @Jordan-Mysten, curious about your opinion on this one.  

My though process is that things like timeouts, and strings for how the user are prompted seem more tied to how the passkey is provided, and could vary by implementation

Reply:
Yeah it seems sensible to keep the signer unaware of this and make the provider for the passkey hold all this config